ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -140,6 +145,8 @@ def main(argv=sys.argv[1:]):
         cmd.extend(['--language={0}'.format(args.language)])
     for include_dir in (args.include_dirs or []):
         cmd.extend(['-I', include_dir])
+    for exclude in (args.exclude or []):
+        cmd.extend(['--suppress=*:', exclude])
```

Comment:
This code is wrong. This is effectively `cppcheck ... --suppress=*: <exclude> ...`, which appends `<exclude>` to the list of files to check but then ignores all checks on *all* files.

What you were looking for was actually `cppcheck ... --suppress=*:<exclude> ...` (without the space before `<exclude>`).